### PR TITLE
[Examples] Update Apache to 2.4.46

### PR DESCRIPTION
**Pull request by mkow**
## Description of the changes 

Seems our Apache builds fail from time to time (see e.g. https://leeroy.cs.unc.edu/job/graphene/5981/console):
```
# work around flex bug
# http://sourceforge.net/tracker/?func=detail&aid=3029024&group_id=97492&atid=618177
perl -0777 -p -i -e 's,\n(void|int) ap_expr_yy[gs]et_column[^\n]+\)\n.*?\n\},,gs' \
	/home/jenkins/workspace/graphene/Examples/apache/httpd-2.4.41/server/util_expr_scan.c
/usr/share/apr-1.0/build/libtool --silent --mode=link x86_64-linux-gnu-gcc  -pthread         -o gen_test_char  gen_test_char.lo 
gawk -f /home/jenkins/workspace/graphene/Examples/apache/httpd-2.4.41/build/make_exports.awk `cat export_files` > exports.c
/home/jenkins/workspace/graphene/Examples/apache/httpd-2.4.41/server/util_expr_scan.c: In function 'ap_expr_yylex':
/home/jenkins/workspace/graphene/Examples/apache/httpd-2.4.41/server/util_expr_scan.c:1617:6: error: unterminated comment
      /* Note: because we've taken care in
      ^
```

After briefly looking at this it seems that the Apache we use has a race condition in its Makefiles - this very file, which looks like being trimmed when used for build is also preprocessed with some `perl` hacks.
And this is definitely a race condition, because the same file downloaded from Jenkins after the failed build isn't trimmed anymore.

For now let's just try updating Apache and hope that they've already fixed the bug.

## How to test this PR? 

Run CI a few times on this PR and see if the bug triggers (OTOH, it was quite rare, so this may not be enough).

---
This change is [Reviewable](https://reviewable.io/reviews/oscarlab/graphene/2193)

### CODE REVIEW

**Review by mkow — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/oscarlab/graphene/2193)* status: 0 of 1 files reviewed, all discussions resolved, not enough approvals from maintainers (2 more required), not enough approvals from different teams (1 more required, approved so far: ITL)

*[a discussion](https://reviewable.io/reviews/oscarlab/graphene/2193#-MUQz4bA7qdxxFn6PrvH:-MUQz4bB9zaf10NzdS7l:b6fgjkz) (no related file):*
Seems we didn't need too many runs to trigger it again :/

https://leeroy.cs.unc.edu/job/graphene-sgx-apps/3088/console:
```
util_expr_scan.c:484:8: error: expected expression at end of input
        48,   48,   48,   48,   48,
        ^
```

I'll need to actually debug it further. But in the meantime I think we can just update this example.

---

**Review by mkow — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/oscarlab/graphene/2193)* status: 0 of 1 files reviewed, all discussions resolved, not enough approvals from maintainers (2 more required), not enough approvals from different teams (1 more required, approved so far: ITL)

*[a discussion](https://reviewable.io/reviews/oscarlab/graphene/2193#-MUQz4bA7qdxxFn6PrvH:-MUR-Bh62IOEaq1hUipp:b-rht0cf) (no related file):*
<details><summary><i>Previously, mkow (Michał Kowalczyk) wrote…</i></summary><blockquote>

Seems we didn't need too many runs to trigger it again :/

https://leeroy.cs.unc.edu/job/graphene-sgx-apps/3088/console:
```
util_expr_scan.c:484:8: error: expected expression at end of input
        48,   48,   48,   48,   48,
        ^
```

I'll need to actually debug it further. But in the meantime I think we can just update this example.
</blockquote></details>

Huh, in this particular build their `perl` hackery run *two* times at once:
```
# work around flex bug
# http://sourceforge.net/tracker/?func=detail&aid=3029024&group_id=97492&atid=618177
perl -0777 -p -i -e 's,\n(void|int) ap_expr_yy[gs]et_column[^\n]+\)\n.*?\n\},,gs' \
	/home/jenkins/workspace/graphene-sgx-apps/Examples/apache/httpd-2.4.46/server/util_expr_scan.c
/usr/share/apr-1.0/build/libtool --silent --mode=link x86_64-linux-gnu-gcc  -pthread         -o gen_test_char  gen_test_char.lo 
gawk -f /home/jenkins/workspace/graphene-sgx-apps/Examples/apache/httpd-2.4.46/build/make_exports.awk `cat export_files` > exports.c
# work around flex bug
# http://sourceforge.net/tracker/?func=detail&aid=3029024&group_id=97492&atid=618177
perl -0777 -p -i -e 's,\n(void|int) ap_expr_yy[gs]et_column[^\n]+\)\n.*?\n\},,gs' \
	/home/jenkins/workspace/graphene-sgx-apps/Examples/apache/httpd-2.4.46/server/util_expr_scan.c
```

Fascinating...

---

**Review by dimakuv — DISMISSED** [review]
> Reviewed 1 of 1 files at r1.
> *[Reviewable](https://reviewable.io/reviews/oscarlab/graphene/2193)* status: all files reviewed, 1 unresolved discussion, not enough approvals from maintainers (1 more required) (waiting on @mkow)

*[a discussion](https://reviewable.io/reviews/oscarlab/graphene/2193#-MUSDNZ59_bycX2sIsf9:-MUSDNZ71yV_gDRGz5OY:b-s4brnn) (no related file):*
Doesn't seem like an important PR for me, but ok.

---

**Review by mkow — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/oscarlab/graphene/2193)* status: all files reviewed, all discussions resolved, not enough approvals from maintainers (1 more required)

*[a discussion](https://reviewable.io/reviews/oscarlab/graphene/2193#-MUSDNZ59_bycX2sIsf9:-MUTam1a7cHpy5yVz_AZ:bgas8vu) (no related file):*
<details><summary><i>Previously, dimakuv (Dmitrii Kuvaiskii) wrote…</i></summary><blockquote>

Doesn't seem like an important PR for me, but ok.
</blockquote></details>

This example was failing pretty often recently (and as you can see, I hit this bug on the first try here :) ), it was worth trying to just update :)

---

**Comment by mkow** [comment]
Jenkins, retest Jenkins-Debug please (Apache build failed, exactly as described...)

**Comment by mkow** [comment]
Jenkins, retest Jenkins-SGX-apps please (Apache build failed, exactly as described...)

**Review by mkow — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/oscarlab/graphene/2193)* status: all files reviewed, all discussions resolved, not enough approvals from maintainers (1 more required)

*[a discussion](https://reviewable.io/reviews/oscarlab/graphene/2193#-MUSDNZ59_bycX2sIsf9:-MUTcXSF2YoSSI8G-X3T:b-9d7lu5) (no related file):*
<details><summary><i>Previously, mkow (Michał Kowalczyk) wrote…</i></summary><blockquote>

This example was failing pretty often recently (and as you can see, I hit this bug on the first try here :) ), it was worth trying to just update :)
</blockquote></details>

Actually, I hit it twice on the first try, on Jenkins-Debug and Jenkins-SGX-apps :)

---

**Review by boryspoplawski — APPROVED** [review]
> Reviewed 4 of 5 files at r2, 2 of 2 files at r4.
> *[Reviewable](https://reviewable.io/reviews/oscarlab/graphene/2193)* status: all files reviewed, all discussions resolved, not enough approvals from maintainers (1 more required), not enough approvals from different teams (1 more required, approved so far: ITL)

*[a discussion](https://reviewable.io/reviews/oscarlab/graphene/2193#-MUTyiQw883aIGpvS7uz:-MUTyiQxEkYUgHzEm-F8:b-6habhz) (no related file):*
I'm taking over this PR, with explicit approval from @mkow

---
___
*[Examples/apache/apache_makefile.patch, line 10 at r4](https://reviewable.io/reviews/oscarlab/graphene/2193#-MUTytRk93q_2_3u-U9W:-MUTytRk93q_2_3u-U9X:bq9mx5r) ([raw file](https://github.com/oscarlab/graphene/blob/698cbba1333335ebacb12b00c48c298d792a0c73/Examples/apache/apache_makefile.patch#L10)):*
> ```Diff
>  #
> -util_expr_scan.c util_expr_parse.c util_expr_parse.h: util_expr_scan.l util_expr_parse.y
> +util_expr_scan.c util_expr_parse.c util_expr_parse.h: util_expr_outputs
> ```

If this proves to solve the issue, I'll try to upstream the patch.

**Comment by boryspoplawski** [comment]
Jenkins, retest this please (need more runs)

**Comment by mkow** [comment]
Jenkins, retest this please (last one successful, let's try a few more)

**Comment by boryspoplawski** [comment]
Jenkins, retest this please (ditto)

**Comment by mkow** [comment]
Jenkins, retest this please (ditto)

**Review by mkow — COMMENTED** [review]
> Reviewed 4 of 5 files at r2, 2 of 2 files at r4.
> *[Reviewable](https://reviewable.io/reviews/oscarlab/graphene/2193)* status: all files reviewed, 1 unresolved discussion, not enough approvals from maintainers (1 more required), not enough approvals from different teams (1 more required, approved so far: ITL)

___
*[Examples/apache/Makefile, line 60 at r4](https://reviewable.io/reviews/oscarlab/graphene/2193#-MUUc4PGBYQEh-Q-SmU0:-MUUc4PHClw6x4FodGP2:bc521fm) ([raw file](https://github.com/oscarlab/graphene/blob/698cbba1333335ebacb12b00c48c298d792a0c73/Examples/apache/Makefile#L60)):*
> ```Makefile
> $(HTTPD_SRC)/configure: $(HTTPD_SRC).tar.gz
> 	tar --touch -xzf $<
> 	cd $(HTTPD_SRC) && patch -p1 -l < ../apache_makefile.patch
> ```

@boryspoplawski: Could you write a short comment why we need this patch?

**Review by boryspoplawski — APPROVED** [review]
> Reviewed 1 of 1 files at r5.
> *[Reviewable](https://reviewable.io/reviews/oscarlab/graphene/2193)* status: all files reviewed, 1 unresolved discussion, not enough approvals from maintainers (1 more required), not enough approvals from different teams (1 more required, approved so far: ITL) (waiting on @mkow)

___
*[Examples/apache/Makefile, line 60 at r4](https://reviewable.io/reviews/oscarlab/graphene/2193#-MUUc4PGBYQEh-Q-SmU0:-MUVK94g-F18iL3OulIU:b-896fix) ([raw file](https://github.com/oscarlab/graphene/blob/698cbba1333335ebacb12b00c48c298d792a0c73/Examples/apache/Makefile#L60)):*
<details><summary><i>Previously, mkow (Michał Kowalczyk) wrote…</i></summary><blockquote>

@boryspoplawski: Could you write a short comment why we need this patch?
</blockquote></details>

Done.

**Review by mkow — COMMENTED** [review]
> Reviewed 1 of 1 files at r5.
> *[Reviewable](https://reviewable.io/reviews/oscarlab/graphene/2193)* status: all files reviewed, all discussions resolved, not enough approvals from maintainers (1 more required), not enough approvals from different teams (1 more required, approved so far: ITL)

**Review by boryspoplawski — APPROVED** [review]
> *[Reviewable](https://reviewable.io/reviews/oscarlab/graphene/2193)* status: all files reviewed, all discussions resolved, not enough approvals from maintainers (1 more required), not enough approvals from different teams (1 more required, approved so far: ITL)

*[a discussion](https://reviewable.io/reviews/oscarlab/graphene/2193#-MUVYAY0DcRtlIQHn8Y5:-MUVYAY0DcRtlIQHn8Y6:bw9crax) (no related file):*
FYI: I've filed a bug report in Apache upstream: https://bz.apache.org/bugzilla/show_bug.cgi?id=65161

---

**Comment by boryspoplawski** [comment]
Jenkins, retest this please (more tests won't hurt, waiting for approval anyway)

**Comment by boryspoplawski** [comment]
Jenkins, retest this please (more tests won't hurt, waiting for approval anyway)

**Comment by boryspoplawski** [comment]
Jenkins, retest this please (more tests won't hurt, waiting for approval anyway)

**Comment by boryspoplawski** [comment]
Jenkins, retest this please (more tests won't hurt, waiting for approval anyway)

**Comment by boryspoplawski** [comment]
Jenkins, retest this please (more tests won't hurt, waiting for approval anyway)

**Comment by boryspoplawski** [comment]
Jenkins, retest this please (more tests won't hurt, waiting for approval anyway)

**Review by dimakuv — APPROVED** [review]
> *[Reviewable](https://reviewable.io/reviews/oscarlab/graphene/2193)* status: :shipit: complete! all files reviewed, all discussions resolved

**Pull request merged**
